lib/bitcoin/BitcoinClient.ts

```diff
@@ -410,23 +458,23 @@ export default class BitcoinClient {
     return [freezeTransaction, freezeScript.toHex()];
   }
 
-  // @ts-ignore
   private async createSpendToFreezeTransaction (
     previousFreezeTransaction: Transaction,
     previousFreezeUntilBlock: number,
-    freezeUntilBlock: number): Promise<Transaction> {
+    freezeUntilBlock: number): Promise<[Transaction, string]> {
 
     const freezeScript = BitcoinClient.createFreezeScript(freezeUntilBlock, this.walletAddress);
     const payToScriptHashOutput = Script.buildScriptHashOut(freezeScript);
     const payToScriptAddress = new Address(payToScriptHashOutput);
 
-    return this.createSpendTransactionFromFrozenTransaction(
+    const freezeTransaction = await this.createSpendTransactionFromFrozenTransaction(

```

Comment:
This is confusing. We are creating a spendTransaction but assign it to const freezeTransaction.

Reply:
Good point .. will update.